### PR TITLE
modtool: Remove comments from C++ iosignatures

### LINKED ISSUE #5222: gr_modtool makeyaml produces some invalid fields
### What happened?

gr_modtool makeyaml produces some invalid fields due to the regex parsing of the impl files

### System Information

OS: Ubuntu 20.10
GR Installation Method: source

### GNU Radio Version

3.10-git (master)

### Specific Version

395-gc3c411be

### Steps to Reproduce the Problem

1) Create a block with gr_modtool
2) gr_modtool makeyaml myblock

The output yaml has a few issues - see log output

### Relevant log output

```shell
id: cudademo_multiply_const
label: multiply const
category: '[Cudademo]'
templates:
  imports: import cudademo
  make: cudademo.multiply_const(${k})
parameters:
- id: k
  label: K
  dtype: float
inputs:
- label: in
  domain: stream
  dtype: raw
  multiplicity: 1 /* max inputs */
outputs:
- label: out
  domain: stream
  dtype: raw
  multiplicity: 1 /*max outputs */
file_format: 1
```

### PR DESCRIPTION
**Pull request by marcusmueller**
## Description

Fix the case where the comments (which are present in gr-newmod) in iosignatures break YAML gen

## Related Issue

Fixes #5222

(At least for reasonable cases. Comments with nested comment symbols, or strings containing C comments, might lead to interesting results. But these are not likely to occur.

## Which blocks/areas does this affect?

gr-modtool

## Testing Done

https://gist.github.com/marcusmueller/f03ff708e0975c82dc80c06156c1ae3b

(can't do automatic tests before we fix modtool to run non-interactively)

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

File: gr-utils/modtool/tools/parser_cc_block.py [hunk 1]
```diff
@@ -78,18 +99,19 @@ def _typestr_to_vlen(typestr):
...
+        except Exception as e:
+            logger.error(f"Error: Can't parse input signature: {e}")
         try:
             iosig['out'] = _figure_out_iotype_and_vlen(iosig_match.group('outcall'),
                                                        iosig_match.group('outtype'))
-            iosig['out']['min_ports'] = iosig_match.group('outmin')
-            iosig['out']['max_ports'] = iosig_match.group('outmax')
+            iosig['out']['min_ports'] = remove_comments(iosig_match.group('outmin'))
+            iosig['out']['max_ports'] = remove_comments(iosig_match.group('outmax'))
         except Exception:
```
Comment (willcode) [inline gr-utils/modtool/tools/parser_cc_block.py:115]: Make same exception format change as above?

File: gr-utils/modtool/tools/parser_cc_block.py [hunk 2]
```diff
@@ -21,6 +21,27 @@ def dummy_translator(the_type, default_v=None):
     return the_type
 
 
+def remove_comments(inp: str) -> str:
+    def comment_removal(match: re.match) -> str:
+        groups = match.groups()
+        if len(groups) == 1:
+            return ''
+        if len(groups) % 3 == 0:
```
Comment (willcode) [inline gr-utils/modtool/tools/parser_cc_block.py:29]: What does this do?

Comment (marcusmueller) [reply gr-utils/modtool/tools/parser_cc_block.py:29]: ah, good catch; that was from a more unified time, when I had a single regex that could rework `2 /* channels per polarizations */ * 2 /* polarizations */` into `2  * 2 ` by replacing a 

    (([not comment])*)((comment)*)(([not comment])*)

match by the not-comment parts. That was to enable nested comment starts, `1 /* /* comment */` being correctly removed, and to disable `"/*comment*"` in strings being removed, but then specialties of python regexes and having to implement the C escaping rules for quotes came in the way. I didn't want to maintain half a C++ tokenizer in a regex, so here I pareto'd and accepted that we'll be able to handle 
`3 /* max inputs */` and `2 /* channels per polarizations */ * 2 /* polarizations */`, but not the problem cases mentioned above. 


The group length check and 3-handling is a remnant of that, gimme a second.

File: gr-utils/modtool/tools/parser_cc_block.py (hunk 1 above)
Comment (marcusmueller) [reply gr-utils/modtool/tools/parser_cc_block.py:115]: :+1:

File: gr-utils/modtool/tools/parser_cc_block.py [hunk 3]
```diff
@@ -21,6 +21,21 @@ def dummy_translator(the_type, default_v=None):
     return the_type
 
 
+def remove_comments(inp: str) -> str:
+    def comment_removal(match: re.match) -> str:
```
Comment (willcode) [inline gr-utils/modtool/tools/parser_cc_block.py:25]: Minor, but how about dropping this code back into the outer function. The subfunc doesn't serve any purpose any more and makes it harder to read.

Comment (marcusmueller) [reply gr-utils/modtool/tools/parser_cc_block.py:25]: not quite sure what you mean here - the callable needs to see `inp`, so it needs to be local

Comment (marcusmueller) [reply gr-utils/modtool/tools/parser_cc_block.py:25]: Made it simpler; no need for the function aside from really pathological corner cases, so omit.

**Comment by willcode** [comment]
@marcusmueller could you squash and sign?

**Pull request merged**